image-rs/src/image.rs

```diff
@@ -229,19 +228,15 @@ impl ImageClient {
                 {
                     Ok(cred) => cred,
                     Err(e) => {
-                        warn!(
-                            "get credential failed, use Anonymous auth instead: {}",
-                            e.to_string()
-                        );
-                        RegistryAuth::Anonymous
+                        bail!("failed to get registry authentication credentials: {:?}", e)
                     }
                 }
             }
             (false, true) => RegistryAuth::Anonymous,
             _ => auth.expect("unexpected uninitialized auth"),
         };
 
-        let mut client = PullClient::new(
+        let mut client: PullClient = PullClient::new(
```

Comment:
do we need this?

Reply:
Hi @Xynnn007 !

Good catch. Probably it was automatically refactoried by vscode. I will revert that.

Reply:
done